packages/@xframework/x/src/core/x.ts

```
@@ -0,0 +1,125 @@
+import type {
+  SyncAdapterCreator,
+  SyncAdapterType,
+  AsyncAdapterCreator,
+  AsyncAdapterType,
+  AdapterRegistry,
+  AdapterResult,
+  BuildOutput,
+} from "./types";
+
+class X<
+  TAdapters extends AdapterRegistry = {},
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)